### DISCUSSION TITLE
FintX 1.0

### DISCUSSION THREAD

**Discussion by namigop** [namigop/FintX#10] [Announcements]
# Release Notes

## Quickstart Guide
[Getting Started](https://github.com/namigop/FintX#readme)

## Features

1. Added support to import and export of request for Unary/Server/Duplex/Client Streaming methods. 
2. Added support to create requests and save into a file
3. Changes to the folder containing the request files are synced to the app. Users can create, edit, delete request files via the OS (Finder, Windows explorer)
4. Copy-Paste (Ctrl/Cmd-C, Ctrl/Cmd-V) support for request *.frqx files
5. Press F2 to rename a request file selected in the explorer tree
6. Added support to reopen the previous session
7. Added support for cancelling reading of response stream (for duplex and server streaming)
8. Display Trailer, Header and Status for async unary calls, client /server/duplex streaming
9. Blocking unary calls are executed in a background thread to avoid blocking the UI
10. Auto-save changes made to open tabs
11.UI Enhancements
    - Placed action controls (start/stop, etc.) in the middle of the tab
    - More consistency in text colors
    -  better icons

## Others
1. Updated windows installer to run with low (non-admin) privilege
2. Multiple bug fixes.
3.  Added more unit tests
12.  Cleaned up compiler warnings

<hr /><em>This discussion was created from the release <a href='https://github.com/namigop/FintX/releases/tag/rel-v1.0'>FintX 1.0</a>.</em>

**Comment by praktanss** [comment]
Really impress with this project UI, i have question that why F# used for Core project and C# for main UI project. Can i create all in c# only?

**Reply by namigop** [reply]
Although It can definitely it can all be in C#, I find that using F# is more suitable for the Core ("the business logic") because it lets you _naturally_ write state-less functional code which I have found to be easy to test, easy to reason about and easy to debug.  Of course these are also achievable in C# but you always have to be on guard to ensure that there is always proper encapsulation.  That takes constant effort in a language where mutability is the default.

As for the main UI project, C# is used in it because a GUI is essentially a big ball of state that needs to be managed. An object-oriented language like C# is a lot more suited for such a task.  I've also experimented with using F# for the GUI (see https://github.com/namigop/OneBella) but I found that writing object-oriented code in a functional-first language felt unnatural.

**Reply by praktanss** [reply]
I appreciate your prompt response, i have checked all your projects on GitHub, really impress with user interface and functionality. I am working on one project for one of my customer and i want to create beautiful stunning sci-fi dynamic UI for the app. I am also thinking about your assistance to write UI if i can and possible with your due concern. My WhatsApp number is +91 9422944380.

One more question is why only Dark Theme for this app? Can we put both Dark and Light option? if yes then How?
